src/osemgrep/core/dune

```diff
@@ -14,6 +14,7 @@
     lwt ; for Project_metadata.ml
     murmur3
     python-str-repr
+    semgrep.reporting
```

Comment:
hmm we don't want src/core to depend on src/reporting.
What is the dependency?
Maybe move this module in src/reporting/ then.

Reply:
the Metavar_replacement is used inside of `match_based_id_partial`. I'll move the whole Semgrep_hashing_functions to src/osemgrep/reporting.